feature/policy_forwarding/GUE/Outer_IPv4/decapsulation/otg_tests/Inner_IP/README.md

```diff
@@ -0,0 +1,337 @@
+    - For inner IPv4 - GUE UDP port 6080
+    - For inner IPv6 - GUE UDP port 6615
+ - Post decapsulation the DUT should copy outer TTL(and decrement) to inner header and maintain the inner DSCP vaule.
+    - If explicit configration is present to not copy the TTL, then it will be honored. 
+ - Decapsulate the packet only if it matches the locally configured GUE VIP and associated UDP port/port-range.
+    - Traffic not subject to match criteria will be forwared using traditional IP forwarding. 
+ - UDP checksum in the transport (outer) header is used for the packet integrity validation, and the DUT will ignore the GUE checksum (non-zero or all-zero). 
+
+## Testbed type
+
+* [`featureprofiles/topologies/atedut_2.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)
+
+## Procedure
+
+### Test environment setup
+
+* Create the following connections:
+* DUT has ingress and egress port connected to the ATE.
+  
+```mermaid
+graph LR; 
+A[ATE:Port1] --Ingress--> B[Port1:DUT:Port2];B --Egress--> C[Port2:ATE];
+```
+
+*  ATE Port 1: Generates GUE-encapsulated traffic with various inner (original) destinations.
+*  ATE Port 2: Receives decapsulated traffic whose inner destination matches the policy.
+  
+### DUT Configuration
+
+1.  Interfaces: Configure all DUT ports as singleton IP interfaces.
+ 
+2.  Static Routes/LSPs:
+    *  Configure an IPv4 static route to GUE decapsulation destination (DECAP-DST) to Null0.
+    *  Have policy configuration that match GUE decapsulation destination and default/non-default GUE UDP port/port-range for the decapsulation.
+       *  If udp port is not configured then the default GUE UDP port will be inherited/used.
+    *  Apply the defined policy on the Ingress (DUT port1) port.
+    *  Configure static routes for encapsulated traffic destinations IPV4-DST1 and IPV6-DST1 towards ATE Port 2.
+    *  Configure static MPLS label binding (LBL1) towards ATE Port 2. Next hop of ATE Port 1 should be indicated for MPLS pop action.
+    *  Configure static routes for destination IPV4-DST2 and IPV6-DST2 towards ATE Port 2.
+
+3.  Policy-Based Forwarding: 
+    *  Rule 1: Match GUE traffic with destination DECAP-DST using destination-address-prefix-set and default/non-default GUE UDP port/port-range for decapsulation.
+      * If udp port is not configured then the default GUE UDP port will be inherited/used.   
+    *  Rule 2: Match all other traffic and forward (no decapsulation).
+    *  Apply the defined policy on the Ingress (DUT port1) port. 
+    
+### PF-1.4.1: GUE Decapsulation of inner IPv4 traffic using default GUE UDP port 6080
+-  Push DUT configuration.
+
+Traffic: 
+-  Generate GUE-encapsulated traffic from ATE Port 1 with destinations matching DECAP-DST and default GUE UDP port 6080.
+-  Inner IPv4 destination should match IPV4-DST1.
+-  Inner-packet DSCP value should be set to 32.
+-  Inner-packet TTL value should be set to 64.
+  
+Verification: 
```

Comment:
Recommend collecting following stats prior to each test and verify after each test. Seems would help to have a comprehensive health check and implications of each of these tests.

Capture the generic health check of the DUT, used modularly in pre/post and during various different tests:
	-	No system/kernel/process/component coredumps
	-	No high CPU spike or usage on control or forwarding plane
	-	No high memory utilization or usage on control or forwarding plane
	-	No processes/daemons high CPU/Memory utilization
	-	No generic drop counters
		- QUEUE drops
			- Interfaces
			- VOQ
		- Fabric drops
		- ASIC drops
	- No flow control frames tx/rx
	- No CRC or Layer 1 errors on interfaces
	- No config commit errors
	- No system level alarms
	- In spec hardware should be in proper state
		- No hardware errors
		- Major Alarms
	- No HW component or SW processes crash


OC paths that can be used:

```
- /system/processes/process/state/cpu-utilization
- /system/processes/process/state/memory-utilization
- /qos/interfaces/interface/input/queues/queue/state/dropped-pkts
- /qos/interfaces/interface/output/queues/queue/state/dropped-pkts
- /qos/interfaces/interface/input/virtual-output-queues/voq-interface/queues/queue/state/dropped-pkts
- /interfaces/interface/state/counters/in-discards
- /interfaces/interface/state/counters/in-errors
- /interfaces/interface/state/counters/in-multicast-pkts
- /interfaces/interface/state/counters/in-unknown-protos
- /interfaces/interface/state/counters/out-discards
- /interfaces/interface/state/counters/out-errors
- /interfaces/interface/state/oper-status
- /interfaces/interface/state/admin-status
- /interfaces/interface/state/counters/out-octets
- /interfaces/interface/state/description
- /interfaces/interface/state/type
- /interfaces/interface/subinterfaces/subinterface/state/counters/in-discards
- /interfaces/interface/subinterfaces/subinterface/state/counters/in-errors
- /interfaces/interface/subinterfaces/subinterface/state/counters/in-unknown-protos
- /interfaces/interface/subinterfaces/subinterface/state/counters/out-discards
- /interfaces/interface/subinterfaces/subinterface/state/counters/out-errors
- /interfaces/interface/ethernet/state/counters/in-mac-pause-frames
- /interfaces/interface/ethernet/state/counters/out-mac-pause-frames
- /interfaces/interface/ethernet/state/counters/in-crc-errors
- /interfaces/interface/ethernet/state/counters/in-block-errors
```

Reply:
We haven't done this in all other tests -- other than core dumps AFAIK. If this is something that we're defining as a 'standard' set of health checks across a subset of tests, let's define them in some common place, with a tight definition of what correctness means.

Reply:
Is there a standard test that depicts the procedure  to collect  the baseline data and validate pre and post test execution? If yes, then we can refer here with the link of the test.